runbenchmark.py

```diff
@@ -140,6 +140,11 @@
     bench.setup(amlb.SetupMode[args.setup])
     if args.setup != 'only':
         res = bench.run(args.task, args.fold)
+        if res is None and args.exit_on_error:
+            #  The desired behavior was to actually have a non-zero exit code.
+            #  I thought this worked before but there are currently so many layers
+            #  of exception handling, I'm not sure how to propagate it back up.
+            raise Exception("No results, likely due to error during evaluation.")
```

Comment:
So the original reason I added the `-e` flag was to allow the benchmark script to exit with a non-zero code, which would allow the failure to be picked up by continuous integration. I am not sure if I hadn't done it correctly at the time, or if the changes since abandoned this behavior.
I realize this is probably not the best way to bring it back, but there are several layers of exception handling so I wasn't sure where or how to go about it. I stuck with this for now until feedback.

Reply:
I'm fixing this separately now: I call return in `finally` blocks in a few places…
I'm confused sometimes with this `finally` behaviour in Python compared with some other languages.

Reply:
see https://github.com/openml/automlbenchmark/pull/297/files
I probably changed the indent when improving/rewriting the job runner a few weeks ago.

Reply:
Will do, thanks! It's hard to tell what is/is not intention at times :)